typescript/utils/src/amount.ts

```diff
@@ -117,3 +118,28 @@ export function convertDecimals(
     return amount.times(new BigNumber(10).pow(difference));
   }
 }
+
+/**
+ * Converts a value with `fromDecimals` decimals to a value with `toDecimals` decimals.
+ * Incurs a loss of precision when `fromDecimals` > `toDecimals`.
+ * @param fromDecimals The number of decimals `value` has.
+ * @param toDecimals The number of decimals to convert `value` to.
+ * @param value The value to convert.
+ * @returns `value` represented with `toDecimals` decimals.
+ */
+export function convertDecimalsEthersBigNumber(
```

Comment:
Yeah, the use of two BigNumber libs is unfortunate but somewhat necessary because Ethers' BigNumber can't handle decimals, making it poorly suited to handling user input. I'm in favor of avoiding Ethers' BN entirely and normalizing to either BigNumber.js or browser native BigInt. The Adapter classes would be a good place to handle that normalization, eventually

Reply:
would you prefer me to move over to BigNumber.js or browser native BigInt now to avoid this fn or plan that for the future?

Reply:
No I don't think that's something you should handle in this PR. 
Create an issue to track: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2751